ui/pages/confirmations/components/confirm/info/approve/spending-cap-context.tsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from 'prop-types';
+import React, { createContext, ReactNode, useContext, useState } from 'react';
+
+const SpendingCapContext = createContext<{
+  customSpendingCap: string;
+  setCustomSpendingCap: (newCap: string) => void;
+} | null>(null);
+
+export const SpendingCapProvider = ({ children }: { children: ReactNode }) => {
+  const [customSpendingCap, setCustomSpendingCap] = useState('');
```

Comment:
Why do we want an entire context just to store a custom spending cap?

Can this state not be encapsulated by whatever component provides the control to mutate it, and then once saved / verified / debounced, we update the actual unapproved transaction in the controller state?

Meaning every other component can simply render the spending cap from the transaction meta and it will be updated if a custom value is set?